src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -742,22 +742,24 @@ public void Init()
                 _sequenceNumber = 0;
             }
 
-            // IsReady returns the current _sequenceNumber, which must be passed to StartAsyncOperation below.
-            public bool IsReady(SocketAsyncContext context, out int observedSequenceNumber)
+            // IsReady returns whether an operation can be executed immediately without queueing it.
+            // When IsReady returns true, the observedSequenceNumber is set.
+            // This ensures the sequence number is read, before executing the operation.
+            // When the operation can't complete, StartAsyncOperation compares against the current
+            // sequence number to see if the operation should be tried again.
+            public bool IsReady(SocketAsyncContext context, out int? observedSequenceNumber)
             {
-                using (Lock())
-                {
-                    observedSequenceNumber = _sequenceNumber;
-                    bool isReady = (_state == QueueState.Ready) || (_state == QueueState.Stopped);
+                QueueState state = (QueueState)Volatile.Read(ref Unsafe.As<QueueState, int>(ref _state));
```

Comment:
QueueState is a byte in size.  This doesn't look safe / correct.
```C#
private enum QueueState : byte
```

Reply:
Oops. I've changed it to `int`.